ros/src/control/scripts/thrust_control.py

```diff
@@ -3,33 +3,94 @@
 from shared_msgs.msg import can_msg, auto_control_msg, thrust_control_msg, thrust_status_msg, thrust_command_msg
 from sensor_msgs.msg import Imu
 from std_msgs.msg import Float32
+from rov import movement/mapper/Complex
+
+desired_a = None
+desired_p = None
+locked_dims_list = None
+disabled_list = None
+inverted_list = None
+auto_ctrl_flag = None
 
 def message_received(msg):
-  # This runs on a seperate thread from the pub
-  pass
+  global desired_a #desired thrust from automatic control
+  global locked_dims_list #locked dimensions
+  desired_a = msg.thrust_vec
+  locked_dims_list = msg.dims_locked
+
+  #initialize flag for automatic control vs pilot control
+  auto_ctrl_flag = 0
+  #if there are locked dimensions, set flag to use automatic control
+  for i in range(0, len(locked_dims_list)):
+    if locked_dims_list[i] == True:
+      auto_ctrl_flag = 1
+
 def command_received(comm):
-  # This runs on a seperate thread from the pub
-  pass
+  global desired_p #desired thrust from pilot
+  global disabled_list #disabled thrusters
+  global inverted_list #inverted thrusters
+  desired_p = comm.desired_thrust
+  disabled_list = comm.disable_thrusters
+  inverted_list = comm.inverted
 
 if __name__ == "__main__":
+  
+  #initialize node and rate
   rospy.init_node('thrust_control')
+  rate = rospy.Rate(10) #10 hz
 
+  #initialize subscribers
   auto_sub = rospy.Subscriber('auto_control', 
       auto_control_msg, message_received)
   comm_sub = rospy.Subscriber('/surface/thrust_command',
       thrust_command_msg, command_received)
-
+  
+  #initialize publishers
   thrust_pub = rospy.Publisher('thrust_control',
-    thrust_control_msg, queue_size=10);
-
+    thrust_control_msg, queue_size=10)
   status_pub = rospy.Publisher('thrust_status',
-    thrust_status_msg, queue_size=10);
+    thrust_status_msg, queue_size=10)
+
+  #initialize auto_ctrl_flag
+  global auto_ctrl_flag
+  auto_ctrl_flag = 0
 
-  rate = rospy.Rate(10) # 10hz
-  # TODO: I2C related activities
   while not rospy.is_shutdown():
-    thrust_pub.publish(thrust_control_msg())
-    status_pub.publish(thrust_status_msg())
-    rate.sleep()
-    
 
+    #set desired thrust to either automatic or pilot control
+    if auto_ctrl_flag == 1:
```

Comment:
We wanted be able to have stabilization working while taking input from the pilot. The point of locked_dims was to tell you which dimensions were being stabilized. Then only those dimensions would be ignored from the pilot, and the stabilization thrust_vect taken in its place. It's not binary for the entire thrust vector like this.